docs/docs/routing.md

```diff
@@ -13,6 +17,40 @@ Gatsby makes it easy to programmatically control your pages. Pages can be create
 
 See the [Creating and Modifying Pages](/docs/creating-and-modifying-pages) for more detail.
 
+When Gatsby creates pages it automatically generates a path to access them. This path will differ depending on how the page was defined.
+
+### Pages defined in `src/pages`
+
+Each `.js` file inside `src/pages` will generate its own page in your Gatsby site. The path for those pages matches the file structure it's found in.
+
+For example, `contact.js` will be found at `yoursite.com/contact`. And `home.js` will be found at `yoursite.com/home`. This works at whatever level the file is created. If `home.js` is moved to a directory called `information`, located inside `src/pages`, the page will now be found at `yoursite.com/information/contact`.
+
+The exception to this rule is any file named `index.js`. Files with this name are matched to the root directory they're found in. That means `index.js` in the root `src/pages` directory is accessed via `yoursite.com`. However, if there is an `index.js` inside the `information` directory, it is found at `yoursite.com/information`.
+
+Note that if no `index.js` file exists in a particular directory that root page does not exist, and attempts to navigate to it will land you on a [404 page](/docs/add-404-page/). For example, `yoursite.com/information/contact` may exist, but that does not guaruntee `yoursite.com/information` exists.
+
+### Pages created with `createPages` hook
+
+Another way to create pages is in your `gatsby-node.js` file using the `createPages` hook. When pages are defined this way the path is explicitly set. For example:
+
+```js:title=gatsby-node.js
+createPage({
+  path: "/routing",
+  component: routing,
+  context: {},
+})
+```
+
+## Conflicting Routes
+
+Since there are multiple ways to create a page, someone may accidentally create multiple pages that are meant to be accessed by the same path. When this happens, Gatsby will show a warning at build time, but the site will still build successfully. In this situation, the page that was built last will be accessible and any other conflicting pages will not be.
```

Comment:
```suggestion
Since there are multiple ways to create a page, different plugins, themes, or sections of code in your `gatsby-node` may accidentally create multiple pages that are meant to be accessed by the same path. When this happens, Gatsby will show a warning at build time, but the site will still build successfully. In this situation, the page that was built last will be accessible and any other conflicting pages will not be.
```

Maybe spelling out the possible perpetrators makes sense here?